### ISSUE TITLE
Docker Desktop on Big Sur hangs on Disk I/O

### ISSUE THREAD

**Issue by mklein201** [docker/for-mac#5414]
[Dockerfile.txt](https://github.com/docker/for-mac/files/6072123/Dockerfile.txt)
![Screen Shot 2021-03-02 at 1 38 18 PM](https://user-images.githubusercontent.com/5229996/109722068-07edc800-7b61-11eb-956f-7e766eea64eb.png)

Attempting to do a make and build on a mounted volume is randomly hanging on blocked disk I/O on Big Sur. It doesn't happen in the same area of the build. Once I get into this state, even `ls` fails to list anything on the mounted volume. These same containers and builds worked fine on Catalina. Once I get into this state, I am unable to even stop and destroy the running containers. The only solution is to restart Docker Desktop.

  - [x ] I have tried with the latest version of Docker Desktop
  - [x ] I have tried disabling enabled experimental features
  - [x ] I have uploaded Diagnostics
  - Diagnostics ID: 8511B889-C7A9-4CC2-B744-1F7DE3DE002F/20210302203749

### Expected behavior
Access to mounted volumes should succeed.

### Actual behavior
Random hangs blocking all disk activity on mounted volumes.

### Information

  - macOS Version: Big Sur 11.2.2
  - Docker Desktop Version: 3.1.0 (51484)

### Steps to reproduce the behavior
The attached Dockerfile is the build image. We are building code for Intel's Software Guard Extensions and need this updated toolchain. The build that is failing uses this container against the Intel SGX repo:
```
git clone https://github.com/intel/linux-sgx.git
cd linux-six.git
git checkout sgx_2.13
make preparation sdk_install_pkg_no_mitigation
prefix=/opt/intel ./linux/installer/bin/sgx_linux_x64_sdk_2.13.100.4.bin
source /opt/intel/sgxsdk/environment
make psw_install_pkg
```
The build sometimes fails in building the `sdk`, and other times gets as far as building the `pew`.

**Comment by mklein201** [comment]
/remove-lifecycle stale.
No one planning to look into this?

**Comment by mgabeler-lee-6rs** [comment]
Broken on 3.5.0 still. I tried changing the mount to `delegated` but that doesn't help. After some later reading, I guess this option does nothing for the gRPC FUSE system. When this happens, I've got no choice but to reboot -- I can't stop the container, I can't even `kill -9` the affected processes from within the containers (I mean I can issue the kill, but the processes don't exit).

My colleagues hitting the same issue have resorted to downgrading to ancient versions ~ 2.4.0 to avoid this issue.

Using that as a hint, I tried switching off the `gRPC FUSE` system and going back to legacy `osxfs`

My test case is pretty simple: run the official [postgres](https://hub.docker.com/_/postgres) container (in my case I use `9.6-alpine`), and bootstrapping a bunch of databases for my company's software stack -- so basically creating a lot of databases and tables, but not really loading any data into them yet. With the stock install, docker can't get through even a dozen of these without hitting this error.

After switching back to the legacy `osxfuse`, everything seems to work fine.
